### ISSUE TITLE
Comment hook auto-reversal unnecessarily retreives old transactions

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#252]
The `WordPoints_Comment_Approved_Hook` base class provides a bootstrap which is currently extended by the Comment Received hook. The Comment hook extends it as well, but it currently disables the auto-reversal behavior discussed here (but see #221).

The `reverse_hook()` method fires each time a comment's status changes, and if it is being disapproved/spammed/trashed it will automatically reverse any transactions which awarded points for that comment.

Consider a case where the comment ends up getting re-approved. The points will be awarded again. But if in the end the comment gets removed again, the reverse hook gets called again to reverse the second transaction. However, when this happens, the log for the first comment approval will be pulled from the database for reversal as well.

Fortunately, the original transaction will not be re-reversed. Only one reversal will be performed per points type, and this will be for the latest transaction because the default ordering for the logs queries is by date descending.

However, we just narrowly avert that disaster. 

As it is, we're still making unnecessary work for ourselves. For a start, we're pulling more data from the database than we need to. This normally won't be a significant strain on anything, unless the comment's status has been toggled thousands of times (unlikely, to say the least). However, we then go on to loop through all of those comments unnecessarily, which just means that we'll be running over and over a `get_comment_meta()` query for metadata that won't exist. Fortunately this is a cached query, so the impact will actually be minimal.

There are several paths forward:

1. Do nothing, because this is sort of an edge case.
1. Refactor the query to only return the latest log for each points type.
1. Add a meta flag in the database on a log once it has been reversed, and query based on that.

Each of these has its issues.

Doing nothing sounds nice, but this code could definitely scare someone. Also, we need to consider that if things get refactored in the future, this could suddenly turn awry. We could add unit tests that would catch this though. And seriously, it is definitely an edge case when it comes to this actually becoming significant.

Refactoring the query to only return the latest log for each type sounds like a simple and straightforward solution that takes little effort and so therefore would be preferable over doing nothing. The only problem is that it would probably be impossible to do this and still use `WordPoints_Points_Logs_Query`.

Adding a meta flag in the database makes refactoring the query easier, but it means that logs which have been reverted prior to this won't have the flag set. That's something that we could just ignore, or we could create a complex update routine to fix it. There's also the consideration that this adds extra fluff to the database. On the other hand, this isn't necessarily fluff, and could actually be useful to us in the future.

In summary, we have an odd somewhat edge-case-y issue here, and the solution could use some more investigation. I would possibly consider refactoring the query if it were possible, but since it probably isn't, I'm currently leaning toward adding the meta flag in the database. I think it would be a nice feature that devs (including moi) would find useful.

**Comment by JDGrimes** [comment]
I've gone with option 3. 

I don't think we'll attempt to do anything about the old logs which don't have this metadata. If we decide to save the reversal's ID as the value of the `auto_reversed` metadata, and if we decide to run an update on the old logs at that point, we'll probably include this as well.
